wolfssl/internal.h

```diff
@@ -3353,20 +3364,32 @@ typedef struct Arrays {
 #define MAX_DATE_SZ 32
 #endif
 
+#define STACK_TYPE_X509    0
+#define STACK_TYPE_NAME    1
+#define STACK_TYPE_BIO     2
+#define STACK_TYPE_OBJ     3
+#define STACK_TYPE_STRING  4
+#define STACK_TYPE_CIPHER  5
+#define STACK_TYPE_ACCESS_DESCRIPTION 6
+
 struct WOLFSSL_STACK {
+    byte type;     /* Set to STACK_TYPE_*. identifies type stored in data.*/
```

Comment:
Placing a one byte member at the top of the structure is not efficient.  Members should be ordered by biggest to smallest.